lib/Parse/ParseType.cpp

```diff
@@ -1676,13 +1674,10 @@ bool Parser::canParseTypeIdentifier() {
 bool Parser::canParseBaseTypeForQualifiedDeclName() {
   BacktrackingScope backtrack(*this);
 
-  // Parse a simple type identifier.
-  if (!canParseSimpleTypeIdentifier())
-    return false;
-
-  // Qualified name base types must be followed by a period.
-  // If the next token starts with a period, return true.
-  return startsWithSymbol(Tok, '.');
+  // Qualified name base types must be able to be parsed,
+  // and must be followed by a period.
+  // If the next token is parasable and starts with a period, return true.
+  return (canParseSimpleTypeIdentifier() && startsWithSymbol(Tok, '.'));
```

Comment:
The old comment here was fine
```suggestion
// Qualified name base types must be followed by a period.		
// If the next token starts with a period, return true.
return (canParseSimpleTypeIdentifier() && startsWithSymbol(Tok, '.'));
```

Reply:
Done